src/docs/development/platform-integration/add-to-app-android/add-flutter-screen.md

```diff
@@ -0,0 +1,266 @@
+being launched from an `OnClickListener`.
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity.createBuilder().build(currentActivity)
+    );
+  }
+});
+```
+
+The above example assumes that your Dart entrypoint is called `main()`, and your initial
+Flutter route is '/'. If you'd like to change either property, use the `IntentBuilder` to 
+provide your desired properties:
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity
+        .createBuilder()
+        .dartEntrypoint("mySpecialEntrypoint")
+        .initialRoute("/my_route")
+        .build(currentActivity)
+      );
+  }
+});
+```
+
+Replace `"mySpecialEntrypoint"` with the name of your desired Dart entrypoint, and replace
+`"/my_route"` with your desired initial route.
+
+### Step 5: (Optional) Use a cached FlutterEngine
+
+Every `FlutterActivity` creates its own `FlutterEngine` by default. Each `FlutterEngine` has
+a non-trivial "warm-up" time. This means that launching a standard `FlutterActivity` comes 
+with a brief delay before your Flutter experience becomes visible. To avoid this delay, you
+can warm-up a `FlutterEngine` before arriving at your `FlutterActivity` and then you can use
+your pre-warmed `FlutterEngine` instead.
+
+To pre-warm a `FlutterEngine`, find a reasonable location in your app to instantiate and hold
+a `FlutterEngine`. For a simplistic app, storing a `FlutterEngine` in the `Application` class
+might be a reasonable option:
+
+```java
+public class MyApplication extends Application {
+  private FlutterEngine flutterEngine;
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+
+    // Initialize Flutter.
+    FlutterMain.startInitialization(this);
+    FlutterMain.ensureInitializationComplete(this, null);
+
+    // Create and warm-up a FlutterEngine.
+    flutterEngine = new FlutterEngine(this);
+
+    DartExecutor.DartEntrypoint entrypoint = new DartExecutor.DartEntrypoint(
+      getResources().getAssets(),
+      FlutterMain.findAppBundlePath(this),
+      "main"
+    );
+
+    flutterEngine.getDartExecutor().executeDartEntrypoint(entrypoint);
+  }
+
+  @NonNull
+  public FlutterEngine getFlutterEngine() {
+    return flutterEngine;
+  }
+}
+```
+
+With a pre-warmed `FlutterEngine`, you now need to instruct your `FlutterActivity` to use the
+pre-warmed `FlutterEngine` instead of creating a new one. To accomplish this, subclass
+`FlutterActivity` and override the `provideFlutterEngine()` method:
+
+```java
+public class MyFlutterActivity extends FlutterActivity {
+  @Override
+  @NonNull
+  protected FlutterEngine provideFlutterEngine(@NonNull Context context) {
+    return ((MyApplication) context.getApplicationContext()).getFlutterEngine();
+  }
+}
+```
+
+Now, when you launch `MyFlutterActivity`, there will be no delay in the display of Flutter
```

Comment:
we can't say 'no delay' :)

'there will be significantly less delay in the display'